packages/devtools_app/test/screens/performance/tabbed_performance_view_test.dart

```diff
@@ -208,7 +244,72 @@ void main() {
     );
 
     testWidgetsWithWindowSize(
-      'only shows Timeline Events tab for non-flutter app',
+      'builds content for Queued Microtasks tab when microtask profiling is '
+      'enabled',
+      windowSize,
+      (WidgetTester tester) async {
+        await tester.runAsync(() async {
+          await setUpServiceManagerWithTimeline(
+            shouldEnableMicrotaskProfiling: true,
+          );
+
+          // First, we verify that instructions are shown in the Queued
+          // Microtasks tab when [controller.queuedMicrotasksController]'s
+          // status is empty.
+
+          final mockQueuedMicrotasksController =
+              controller.queuedMicrotasksController
+                  as MockQueuedMicrotasksController;
+          when(mockQueuedMicrotasksController.status).thenReturn(
+            const FixedValueListenable<QueuedMicrotasksControllerStatus>(
+              QueuedMicrotasksControllerStatus.empty,
+            ),
+          );
+
+          await pumpView(tester);
+
+          expect(find.byType(AnalyticsTabbedView), findsOneWidget);
+          expect(find.byType(DevToolsTab), findsNWidgets(4));
+
+          await tester.tap(find.text('Queued Microtasks'));
+          await tester.pumpAndSettle();
+
+          expect(find.byType(RefreshQueuedMicrotasksButton), findsOneWidget);
+          expect(find.byType(QueuedMicrotasksTabView), findsOneWidget);
+          expect(
+            find.byType(RefreshQueuedMicrotasksInstructions),
+            findsOneWidget,
+          );
+
+          // Then, we verify that details about the queued microtasks are shown
+          // when [controller.queuedMicrotasksController]'s status is ready.
+
+          when(mockQueuedMicrotasksController.status).thenReturn(
+            const FixedValueListenable<QueuedMicrotasksControllerStatus>(
+              QueuedMicrotasksControllerStatus.ready,
+            ),
+          );
+          when(
+            mockQueuedMicrotasksController.selectedMicrotask,
+          ).thenReturn(FixedValueListenable(testMicrotask));
+          when(
+            mockQueuedMicrotasksController.queuedMicrotasks,
+          ).thenReturn(FixedValueListenable(testQueuedMicrotasks));
+
+          await tester.tap(find.text('Timeline Events'));
+          await tester.pumpAndSettle();
+          await tester.tap(find.text('Queued Microtasks'));
```

Comment:
why do we have to switch tabs? we should just be able to call pumpAndSettle to trigger a rebuild, and if that doesn't work then maybe there is a bug?

Reply:
The problem was that I was trying to use `when` to modify the value of `mockQueuedMicrotasksController.status`, but I should have made `mockQueuedMicrotasksController.status` a `ValueNotifier` and used that to change the value.